ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+
+The annotations are:
+
+```
+ingress.kubernetes.io/auth-type:[basic|digest]
+```
+
+Indicates the [HTTP Authentication Type: Basic or Digest Access Authentication](https://tools.ietf.org/html/rfc2617).
+
+```
+ingress.kubernetes.io/auth-secret:secretName
+```
+
+Name of the secret that contains the usernames and passwords with access to the `path/s` defined in the Ingress Rule.
+The secret must be created in the same namespace than the Ingress rule
+
+```
+ingress.kubernetes.io/auth-realm:"realm string"
+```
+
+Please check the [auth](examples/custom-upstream-check/README.md) example
+
+
+### Rewrite
+
+In some scenarios the exposed URL in the backend service differs from the specified path in the Ingress rule. Without a rewrite any request will return 404.
+To allow this is possible to use the annotation `ingress.kubernetes.io/rewrite-target` with the path expected in the service.
+If the application contains relative links is possible to add an additional annotation `ingress.kubernetes.io/add-base-url` that will append a `base` tag in the header of the returned HTML from the backend.
+
+
+Please check the [rewrite](examples/rewrite/README.md) example
+
+
+### Rate limiting
+
+The annotations `ingress.kubernetes.io/limit-connections` and `ingress.kubernetes.io/limit-rps` allows the creation of a limit in the connections that can be opened by a single client IP address. This can be use to mitigate [DDoS Attacks](https://www.nginx.com/blog/mitigating-ddos-attacks-with-nginx-and-nginx-plus)
+
+`ingress.kubernetes.io/limit-connections`: number of concurrent allowed connections from a single IP address
+
+`ingress.kubernetes.io/limit-rps`: number of allowed connections per second from a single IP address
+
+
+Is possible to specify both annotation in the same Ingress rule. In this case the order of the limit is `limit-connections` and then `limit-rps`
+
+
+### Secure NGINX upstreams
+
+By default NGINX uses `http` to reach the services. Adding the annotation `ingress.kubernetes.io/secure-upstream: "true"` in the ingress rule changes the protocol to `https`.
+
+
+### Whitelist source range
+
+Using the annotation `ingress.kubernetes.io/whitelist-source-range` is possible to specify one or ranges of client IP addresses from which is possible to access the `Path` in the Ingress rule, e.g. `10.0.0.0/24,172.10.0.1`
+For a global restriction (any URL) is possible to use `whitelist-source-range` in the NGINX config map
+
+*Note:* adding an annotation overrides any global restriction
+
+Please check the [whitelist](examples/whitelist/README.md) example
+
+
+
+### **Allowed parameters in configuration config map:**
+
+##### *body-size:*
+
+Sets the maximum allowed size of the client request body. See NGINX [client_max_body_size](http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size)
```

Comment:
Can you just put these on the same line?

__body size__: Sets the maximum allowed